website/next.config.mjs

```diff
@@ -1,68 +1,75 @@
-      versions.destinations[name] = latest.split('-')[3]
-    } else if (dirent.isFile() && dirent.name == "cli.json") {
-      versions.cli = JSON.parse(fs.readFileSync('./versions/' + dirent.name, 'utf8')).latest.split('-')[1]
-    }
+  return parts[1];
+}
+
+function parseName(name) {
+  const parts = name.split("-");
+  if (parts.length === 2) {
+    return parts[1];
   }
-  dir.closeSync()
-  return versions
+  return parts[0];
+}
+
+function getVersionsForPrefix(prefix, files) {
+  return Object.fromEntries(
+    files
+      .filter((file) => file.name.split("-")[0] == prefix)
+      .map((file) => [parseName(file.name), parseVersion(file.latest)])
+  );
+}
+
+function getVersions() {
+  const files = fs
+    .readdirSync("./versions", { withFileTypes: true })
+    .filter((dirent) => dirent.isFile())
+    .map((file) => ({
+      name: path.basename(file.name, path.extname(file.name)),
+      latest: JSON.parse(fs.readFileSync(`./versions/${file.name}`, "utf8"))
+        .latest,
+    }));
+
+  return {
+    cli: getVersionsForPrefix("cli", files),
+    sources: getVersionsForPrefix("source", files),
+    destinations: getVersionsForPrefix("destination", files),
+  };
 }
 
-const versions = getVersions()
+const versions = getVersions();
+console.log(versions);
 
 const replaceMdxCodeVersions = (node) => {
-  if (node.type === 'text') {
-    let match = node.value.match(reSourcePluginVersion)
-    if (match && match.length >= 1) {
-      let version = versions.sources[match[1].toLowerCase()]
-      if (version === undefined) {
-        throw new Error(`Could not find version for source plugin ${match[1]}`)
+  if (node.type === "text") {
+    Object.entries(patterns).forEach(([key, pattern]) => {
+      const match = node.value.match(pattern);
+      if (match && match.length >= 1) {
+        const version = versions[key][match[1].toLowerCase()];
+        if (version === undefined) {
+          throw new Error(`Could not find version for ${key} ${match[1]}`);
+        }
+        node.value = node.value.replace(pattern, version);
       }
-      node.value = node.value.replace(reSourcePluginVersion, version)
-    }
-    match = node.value.match(reDestPluginVersion)
-    if (match && match.length >= 1) {
-      let version = versions.destinations[match[1].toLowerCase()]
-      if (version === undefined) {
-        throw new Error(`Could not find version for destination plugin ${match[1]}`)
-      }
-      node.value = node.value.replace(reDestPluginVersion, version)
-    }
-    if (node.value.includes(reCLI)) {
-      let version = versions.cli
-      if (version === undefined) {
-        throw new Error(`Could not find version for cli ${match}`)
-      }
-      node.value = node.value.replace(reCLI, version)
-    }
+    });
   }
   if (node.children !== undefined) {
-    node.children.map(replaceMdxCodeVersions)
+    node.children.forEach(replaceMdxCodeVersions);
```

Comment:
`map` is indented to map the array into another array. I think `forEach` is what we want here